src/QMCHamiltonians/ForceBase.h

```diff
@@ -79,45 +57,50 @@ struct ForceBase
   void setObservablesStress(QMCTraits::PropertySetType& plist);
   void setParticleSetF(QMCTraits::PropertySetType& plist, int offset);
   void setParticleSetStress(QMCTraits::PropertySetType& plist, int offset);
-};
 
-struct BareForce : public OperatorBase, public ForceBase
-{
-private:
-  const int d_ei_ID;
+  bool getAddIonIon() const noexcept;
+  void setAddIonIon(bool val) noexcept;
 
-public:
-  BareForce(ParticleSet& ions, ParticleSet& elns);
-  std::string getClassName() const override { return "BareForce"; }
-  void resetTargetParticleSet(ParticleSet& P) override;
+  const ParticleSet::ParticlePos& getForces() const noexcept;
+  void setForces(const ParticleSet::ParticlePos& forces);
+  void setForces(Real val);
 
-  Return_t evaluate(ParticleSet& P) override;
+  const ParticleSet::ParticlePos& getForcesIonIon() const noexcept;
```

Comment:
To my compiler knowledge, having accessor functions defined in the header files allows being in-lined by compilers. Usually they are just simple return and having them defined in the header file helps readability.
I'm not aware of any advantages of having the definition in cpp.

Reply:
Simple accessor functions should be kept in header files for this reason. 

[ The source base has a good number of large functions headers that could be relocated, but small and potentially important-for-performance accessors should be kept. We do not want to have to rely on e.g. LTO for efficiency.]

Reply:
My two cents is that in general inlining is not good for caching and bloats the executable, keeping implementation in ccp make translation units confined without compromising performance. Not a problem for QMCPACK users, but for developers modifying header implementations bloat compilation times, depending on the header is not uncommon that ~ 300 targets need to be rebuilt adding little benefit.

Reply:
Happily, should we arrive at a debate, this is testable.

Reply:
> My two cents is that in general inlining is not good for caching and bloats the executable, keeping implementation in cpp make translation units confined without compromising performance. Not a problem for QMCPACK users, but for developers modifying header implementations bloat compilation times, depending on the header is not uncommon that ~ 300 targets need to be rebuilt adding little benefit.

True. I only expect accessor functions defined in header files
```
const ParticleSet::ParticlePos& getForcesIonIon() const noexcept
{
  return forces_ion_ion_; // return a member.
}

Reply:
Please put all the get function definition in the header file and I will approve this PR.

Reply:
One liners should be fine, not sure if the time critical [aspect](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f5-if-a-function-is-very-small-and-time-critical-declare-it-inline) applies, but readability is a good point.